packages/lit-analyzer/src/cli/analyze-command.ts

```diff
@@ -111,7 +111,11 @@ export async function analyzeCommand(globs: string[], config: LitAnalyzerCliConf
 		printText(statsText, config);
 	}
 
-	return stats.diagnostics <= (config.maxWarnings || 0) || config.maxWarnings === -1;
+	if (stats.errors !== 0) {
```

Comment:
you'll probably also want to update the `failFast` condition further up to only fail if we have errors or have set `maxWarnings` to less than how many warnings were seen so far

Reply:
As the documentation says:

>Exit the process right after the first problem has been found

I think the current behavior is okey.

Reply:
I agree with @43081j, - I'll implement that after merging this PR. The word `problem` in that context was meant to be interpreted as "the first diagnostic that results in a failure", so I think it would make sense to take `--maxWarnings` into account for the `--failFast` option.